Makefile

```
@@ -24,38 +24,16 @@ help:
 	@echo ""
 	@echo "  Variables"
 	@echo ""
-	@echo "    OLENA_VERSION  Olena version to use ('$(OLENA_VERSION)')"
 	@echo "    PREFIX         directory to install to ('$(PREFIX)')"
 	@echo "    PYTHON         Python binary to bind to ('$(PYTHON)')"
 	@echo "    PIP            Python pip to install with ('$(PIP)')"
 
 # END-EVAL
 
-# Olena version to use
-#OLENA_VERSION ?= 2.1
-OLENA_VERSION ?= git
+OLENA_DIR = olena-git
 
-OLENA_DIR = olena-$(OLENA_VERSION)
-OLENA_TARBALL = $(OLENA_DIR).tar.gz
-
-$(OLENA_TARBALL):
-	wget -N https://www.lrde.epita.fr/dload/olena/$(OLENA_VERSION)/$(OLENA_TARBALL)
-
-$(OLENA_DIR): olena-configure-python3.patch
-$(OLENA_DIR): olena-disable-doc.patch
-$(OLENA_DIR): olena-add-bin-negate-toggles.patch
-ifeq ($(OLENA_VERSION),git)
-$(OLENA_DIR):
-	git clone https://gitlab.lrde.epita.fr/olena/olena.git $@
-else
-$(OLENA_DIR): olena-configure-boost.patch
-$(OLENA_DIR): olena-fix-magick-load-catch-exceptions.patch
-$(OLENA_DIR): $(OLENA_TARBALL)
-	tar zxf $(OLENA_TARBALL)
-endif
-	for patch in $(filter %.patch, $^); do \
-		patch -N -d $(OLENA_DIR) -p0 < $$patch; \
-	done
+$(OLENA_DIR)/configure:
```

This should not be unconditional, because it would only run the recipe once and never update or reconfigure thereafter. (So this would only work with fresh clones.)
```suggestion
$(OLENA_DIR)/configure: assets-update
```

---

Makes sense, I'll add a comment to make it clear why this non-obvious required target.

---

You could also rename that phony target from `assets-update` to `always-update` or something.

---

I just amended the comment on the `assets-update` target. It's not complex makefile magic, I think this suffices.